NitroxClient/GameLogic/Entities.cs

```diff
@@ -66,17 +66,17 @@ public void Spawn(List<Entity> entities)
             }
         }
 
-        private EntityCell EnsureCell(Entity entity)
+        private EntityCell EnsureCell(Entity entity, Optional<GameObject> liveRoot)
         {
             EntityCell entityCell;
 
-            if (!EntityCells.TryGetValue(entity.AbsoluteEntityCell, out entityCell))
+            if (!entityCells.TryGetValue(entity.AbsoluteEntityCell, out entityCell) && liveRoot.IsPresent() && liveRoot.Get().GetComponent<LargeWorldEntityCell>())
             {
                 Int3 batchId = ToInt3(entity.AbsoluteEntityCell.BatchId);
                 Int3 cellId = ToInt3(entity.AbsoluteEntityCell.CellId);
                 entityCell = new EntityCell(LargeWorldStreamer.main.cellManager, LargeWorldStreamer.main, batchId, cellId, entity.Level);
-                EntityCells.Add(entity.AbsoluteEntityCell, entityCell);
-                entityCell.EnsureRoot();
+                entityCells.Add(entity.AbsoluteEntityCell, entityCell);
+                entityCell.liveRoot = liveRoot.Get();
                 entityCell.liveRoot.name = string.Format("CellRoot {0}, {1}, {2}; Batch {3}, {4}, {5}", cellId.x, cellId.y, cellId.z, batchId.x, batchId.y, batchId.z);
```

Comment:
Should we remove this line now that we are calling liveRoot.get()?

Reply:
Im assuming you mean the name line?